### PR TITLE
Bundle layer request

### PR DESCRIPTION
**Pull request by annarieger**
### Breaking change

* Fetch all configured application layers at once via `allLayersByIds` query
* Add some tests and null checks

Introduces breaking change due to modification of `parseNodes` and `parseFolder` method arguments.

Both methods are expecting a mandatory array of layer entities as second argument.

Old signature:
```
parseNodes(nodes: DefaultLayerTree[], projection?: OlProjectionLike)
parseFolder(nodes: DefaultLayerTree[], projection?: OlProjectionLike)
```

New signature: 
```
parseNodes(nodes: DefaultLayerTree[], layers: Layer[], projection?: OlProjectionLike)
parseFolder(nodes: DefaultLayerTree[], layers: Layer[], projection?: OlProjectionLike)
```
Please review @terrestris/devs

### CODE REVIEW

File: src/graphqlqueries/Layers.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export const allLayersByIds = `query($ids: [Int]) {
+  allLayersByIds(ids: $ids) {
```
Comment (LukasLohoff) [inline src/graphqlqueries/Layers.ts:2]: wouldn't `layersByIds` or `layersByIdList` make more sense? since this only returns layers in `$ids`

**Review by LukasLohoff — APPROVED** [review]

File: src/graphqlqueries/Layers.ts (hunk 1 above)
Comment (LukasLohoff) [reply src/graphqlqueries/Layers.ts:2]: :+1:

File: src/parser/SHOGunApplicationUtil.ts [hunk 2]
```diff
@@ -36,6 +36,10 @@ import SHOGunAPIClient from '../service/SHOGunAPIClient';
 
 import { getBearerTokenHeader } from '../security/getBearerTokenHeader';
 
+
+import {
+  allLayersByIds
+} from '../graphqlqueries/Layers';
```
Comment (dnlkoch) [inline src/parser/SHOGunApplicationUtil.ts:42]:
```suggestion
import {
  allLayersByIds
} from '../graphqlqueries/Layers';

```

File: src/parser/SHOGunApplicationUtil.ts [hunk 3]
```diff
@@ -95,49 +99,75 @@ class SHOGunApplicationUtil<T extends Application, S extends Layer> {
...
+      Logger.warn('Cannot parse the layers in layertree because no ' +
+        'SHOGunClient has been provided.');
+      return;
+    }
 
-    const tree = new OlLayerGroup({
-      layers: nodes.reverse(),
-      visible: layerTree.checked
-    });
+    let applicationLayerIds: number[] = this.getLayerIds(layerTree.children);
```
Comment (dnlkoch) [inline src/parser/SHOGunApplicationUtil.ts:108]:
```suggestion
    let layerIds: number[] = this.getLayerIds(layerTree.children);
```

File: src/service/GraphQLService/index.ts [hunk 4]
```diff
@@ -33,7 +33,7 @@ export class GraphQLService {
     this.keycloak = opts.keycloak;
   }
 
-  async sendQuery<T>(query: GraphQLQueryObject, fetchOpts?: RequestInit): Promise<GraphQLResponse<T>> {
+  async sendQuery<T>(query: GraphQLQueryObject, fetchOpts?: RequestInit): Promise<any> {
```
Comment (dnlkoch) [inline src/service/GraphQLService/index.ts:36]: Why `any`?

**Review by dnlkoch — COMMENTED** [review]
Thanks for this @arieger!

Please note that for automated deployments the commits should follow the semantic-release commit naming conventions.

File: src/service/GraphQLService/index.ts (hunk 4 above)
Comment (annarieger) [reply src/service/GraphQLService/index.ts]: Just forgot to change it back :) Fixed in 5671c02
